norfair/metrics.py

```diff
@@ -0,0 +1,327 @@
+import os
+import time
+import sys
+import numpy as np
+import random
+import cv2
+from rich import print
+from .utils import validate_points
+from rich.progress import BarColumn, Progress, TimeRemainingColumn, track
+from norfair import Detection
+import norfair
+import motmetrics as mm
+from collections import OrderedDict
+import pandas as pd
+
+class InformationFile:
+    def __init__(self, file_path):
+        with open(file_path, "r") as myfile:
+            self.file = myfile.read()
+
+    def search(self, variable_name):
+        index_position_on_this_document = self.file.find(variable_name)
+        index_position_on_this_document = index_position_on_this_document + len(variable_name)
+        while not self.file[index_position_on_this_document].isdigit():
+            index_position_on_this_document += 1
+        value_string = ""
+        while self.file[index_position_on_this_document].isdigit():
+            value_string = value_string + self.file[index_position_on_this_document]
+            index_position_on_this_document += 1
+        return int(value_string)
+
+def write_predictions(frame_number, objects=None, output_text_file=None):
+    '''
+    Write tracked object information on the output file (for this frame), in the format
+    frame_number, id, bb_left, bb_top, bb_width, bb_height, -1, -1, -1, -1
+    '''
+    # write tracked objects information in the output file
+    for t in range(len(objects)):
+        frame_str = str(int(frame_number))
+        id_str = str(int(objects[t].id))
+        bb_left_str = str((objects[t].estimate[0, 0]))
+        bb_top_str = str((objects[t].estimate[0, 1]))  # [0,1]
+        bb_width_str = str((objects[t].estimate[1, 0] - objects[t].estimate[0, 0]))
+        bb_height_str = str((objects[t].estimate[1, 1] - objects[t].estimate[0, 1]))
+        row_text_out = (
+            frame_str
+            + ","
+            + id_str
+            + ","
+            + bb_left_str
+            + ","
+            + bb_top_str
+            + ","
+            + bb_width_str
+            + ","
+            + bb_height_str
+            + ",-1,-1,-1,-1"
+        )
+        output_text_file.write(row_text_out)
+        output_text_file.write("\n")
+
+
+def process_text_file(text_file_path):
+    """
+    This function proccess detections and ground truth text files, ordering the data by frame, and making the box coordinates reference to corners positions.
+    """
+    matrix_data = np.loadtxt(text_file_path, dtype="f", delimiter=",")
+    row_order = np.argsort(matrix_data[:, 0])
+    matrix_data = matrix_data[row_order]
+    # coordinates refer to box corners
+    matrix_data[:, 4] = matrix_data[:, 2] + matrix_data[:, 4]
+    matrix_data[:, 5] = matrix_data[:, 3] + matrix_data[:, 5]
+    return matrix_data
+
+
+class PredictionsTextFile:
```

Comment:
Please add a docstring describing what this class does and what each of its arguments do.